convert_full_model.py

```diff
@@ -1,165 +1,127 @@
 
-y_pred = model.predict(X_pre)
-y_hls = hls_model.predict(X_pre)
+hls_model.compile()
 
-met = np.hypot(y[:, 0], y[:, 1])
-met_pred = np.hypot(y_pred[:, 0], y_pred[:, 1]) * normFac
-met_hls = np.hypot(y_hls[:, 0], y_hls[:, 1]) * normFac
-met_pup_x = np.sum(X[:, :, 1], axis=-1)
-met_pup_y = np.sum(X[:, :, 2], axis=-1)
-met_pup = np.hypot(met_pup_x, met_pup_y)
+if build_option['synth']:
+    hls_model.build(**build_option)
+    hls4ml.report.read_vivado_report(output_dir)
 
-import seaborn
-import pandas as pd
-import matplotlib.pyplot as plt
-
-df = pd.DataFrame.from_dict({'Gen MET': met, 'PUPPI MET': met_pup, 'QKeras MET': met_pred, 'hls4ml MET': met_hls})
-plt.figure()
-seaborn.pairplot(df, corner=True)
-plt.savefig(f'{output_dir}/profiling_MET.png', dpi=300)
-
-df = pd.DataFrame.from_dict({'Gen MET x': y[:, 0], 'PUPPI MET x': met_pup_x, 'QKeras MET x': y_pred[:, 0], 'hls4ml MET x': y_hls[:, 0]})
-plt.figure()
-seaborn.pairplot(df, corner=True)
-plt.savefig(f'{output_dir}/profiling_MET_x.png', dpi=300)
-
-df = pd.DataFrame.from_dict({'Gen MET y': y[:, 1], 'PUPPI MET y': met_pup_y, 'QKeras MET y': y_pred[:, 1], 'hls4ml MET y': y_hls[:, 1]})
-plt.figure()
-seaborn.pairplot(df, corner=True)
-plt.savefig(f'{output_dir}/profiling_MET_y.png', dpi=300)
-
-response_pup = met_pup / met
-response_pred = met_pred / met
-response_hls = met_hls / met
-bins = np.linspace(0, 2, 25)
-plt.figure(figsize=(12, 5))
-plt.subplot(1, 3, 1)
-plt.hist(response_pup, bins=bins, label=f'PUPPI, median={np.median(response_pup):0.2f}, IQR={scipy.stats.iqr(response_pup):0.2f}')
-plt.legend()
-plt.xlabel("MET response $\hat{y}/y$")
-plt.ylabel("Events")
-plt.subplot(1, 3, 2)
-plt.hist(response_pred, bins=bins, label=f'QKeras, median={np.median(response_pred):0.2f}, IQR={scipy.stats.iqr(response_pred):0.2f}')
-plt.legend()
-plt.xlabel("MET response $\hat{y}/y$")
-plt.ylabel("Events")
-plt.subplot(1, 3, 3)
-plt.hist(response_hls, bins=bins, label=f'hls4ml, median={np.median(response_hls):0.2f}, IQR={scipy.stats.iqr(response_hls):0.2f}')
-plt.legend()
-plt.xlabel("MET response $\hat{y}/y$")
-plt.ylabel("Events")
-plt.tight_layout()
-plt.savefig(f"{output_dir}/response_MET.png", dpi=300)
-
-y_hls, hls4ml_trace = hls_model.trace(X_pre)
-keras_trace = hls4ml.model.profiling.get_ymodel_keras(model, X_pre)
-
-for layer in hls4ml_trace.keys():
-    plt.figure()
-    if layer not in keras_trace: continue
-    plt.scatter(hls4ml_trace[layer].flatten(), keras_trace[layer].flatten(), s=0.2)
-    min_x = min(np.amin(hls4ml_trace[layer]), np.amin(keras_trace[layer]))
-    max_x = max(np.amax(hls4ml_trace[layer]), np.amax(keras_trace[layer]))
-    plt.plot([min_x, max_x], [min_x, max_x], c='gray')
-    plt.xlabel(f'hls4ml {layer}')
-    plt.ylabel(f'QKeras {layer}')
-    plt.savefig(f'{output_dir}/profiling_{layer}.png', dpi=300)
+#f = h5py.File('../L1METML/data/test_data.h5')
```

Comment:
why is this code commented?

if it's too slow, you can make it an option to run.

if it should be removed and put into a different script, do that.